daemon/mgr/container_types.go

```diff
@@ -34,7 +34,27 @@ type ContainerRemoveOption struct {
 // Container represents the container instance in runtime.
 type Container struct {
 	sync.Mutex
-	meta *types.ContainerInfo
+	meta       *ContainerMeta
+	DetachKeys string
+}
+
+// ContainerMeta wrap ContainerInfo and implement meta.Object interface.
+type ContainerMeta types.ContainerInfo
+
+// Key returns container's id.
+func (cm *ContainerMeta) Key() string {
+	return cm.ID
+}
+
+// ToContainerInfo converts ContainerMeta to ContainerInfo
+func (cm *ContainerMeta) ToContainerInfo() *types.ContainerInfo {
```

Comment:
We can use "ContainerMeta" type directly in our core codes, so don't need to transform ContainerMeta to ContainerInfo by "ToContainerInfo"

Reply:
Actually ctrd uses ContainerInfo as well. So in a follow-up PR, I will try to fix this. Could we make this move on?

Since the conflict thing bothers a lot. 😄

Reply:
I think it's ok.